qkernel/src/interrupt/aarch64/mod.rs

```diff
@@ -1,6 +1,141 @@
+use super::super::syscall_dispatch_aarch64;
 pub unsafe fn InitSingleton() {
 }
 
 pub fn init() {
-    // TODO load IDT
-}
\ No newline at end of file
+    // TODO set up GIC
+}
+
+use core::arch::asm;
+
+#[repr(C)]
+pub struct ExceptionStateEL1{
+    pub spsr_el1: u64,
+    pub elr_el1: u64,
+    pub esr_el1: u64,
+    pub tpidr_el0: u64,
+    pub x: [u64;32]
```

Comment:
Why are there 32 general registers? isn't it 31? I don't see where to save x31 in `el0_sync_save_regs`, and we don't save sp because we handle exception on user stack?